ddtrace/ext/ci.py

```diff
@@ -41,28 +43,33 @@
 
 
 def _normalize_ref(name):
+    # type: (Optional[str]) -> Optional[str]
     return _RE_TAGS.sub("", _RE_ORIGIN.sub("", _RE_REFS.sub("", name))) if name is not None else None
 
 
 def _filter_sensitive_info(url):
+    # type: (Optional[str]) -> Optional[str]
     return _RE_URL.sub("\\1", url) if url is not None else None
 
 
 def tags(env=None):
+    # type: (Optional[MutableMapping[str, str]]) -> MutableMapping[str, str]
+    """Extract and set tags from provider environ."""
     env = os.environ if env is None else env
-    tags = {}
+    tags = {}  # type: MutableMapping[str, Optional[str]]
     for key, extract in PROVIDERS:
         if key in env:
             tags = extract(env)
             break
 
+    git_info = git.extract_git_metadata(tags)
```

Comment:
```suggestion
    git_info = git.extract_git_metadata()
```

I think we should have `extract_git_metadata` only return relevant tags that it was able to extract.